# dbVar Structural Variant (SV) Datasets

## ****This is a work in progress and is subject to change****

Advances in genomic technologies have revealed structural variations (SV) to be prevalent in all human DNA, and subsequent research has increasingly implicated SV in phenotypic diversity and disease. In the next few years millions of genomes will be sequenced, leading to the discovery of many millions of SVs that will need to be analyzed to understand their functional impacts. A critical step in variant analysis and interpretation is the comparison of SVs found in an individual or population with known variants extant in public databases such as dbVar. This "case-control"-like comparison is needed to gain biological insight into novel variants from existing variants and their annotations. To address this need, dbVar proposes the creation of structural variation reference datasets of known SV, complete with rich genomic and biological annotations, for use in novel SV analysis, annotaion, and related workflows.

dbVar is a NCBI database of human genomic structural variation with more than 5 million submitted SVs from 157 human studies. These include data from large diversity projects such as the 1000 Genomes Project and a global population CNV survey (Sudmant et al., 2015), as well as from clinical resources such as ClinVar and ClinGen. Using this large collection of variants, we created several non-redundant SV datasets, separated according to variant type. We then annotated these datasets using available information on genes, molecular consequence, clinical significance, dosage sensitivity, and regulatory regions, and on relevant genomic structural features such as repetitive regions, segmental duplications, and assembly-assembly alignment anomalous regions. These reference data and annotations are intended to facilitate the integration and comparison of dbVar SV data with other genome annotations (such as disease phenotype and population frequencies) and to provide insights into the impact of the SV on biological functions.

The resulting alpha-release datasets are currently available in tab-delimited formats at https://github.com/dbvar/Sandbox/Non-redundant_Structural_Variants. Our goal is to update these files on a regular basis incorporating new variant submissions, genomic features, Gene, RefSeq, ClinVar and other annotation information as it becomes available. We encourage users to test these files and to provide feedback, either by emailing dbvar@ncbi.nlm.nih.gov or by submitting a request directly at https://github.com/ncbi/dbvar/issues.

============================
